### ISSUE TITLE
Error when pasting a blueprint on a pole

### ISSUE THREAD

**Issue by Wesmania** [Touhma/DSP_Plugins#143]
I got this stack trace when placing a bunch of solar panels on a pole:
```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at GalacticScale.PatchOnPlanetGrid.DetermineLongitudeSegmentCount (PlanetGrid __instance, System.Int32 latitudeIndex, System.Int32 segment, System.Int32& __result) [0x0007b] in <91fc5bf28e8c4542a4f8d2262c114014>:0 
  at PlanetGrid.DetermineLongitudeSegmentCount (System.Int32 latitudeIndex, System.Int32 segment) [0x00005] in <ee3f7728a4054543a91656140ca60f7c>:0 
  at BlueprintUtils.GetLongitudeSegmentCount (System.Int32 _latitudeGridIdx, System.Int32 _segmentCnt) [0x00011] in <ee3f7728a4054543a91656140ca60f7c>:0 
  at BlueprintUtils.SnapTropic (PlayerAction_Build _actionBuild, BlueprintData _blueprintData, UnityEngine.Vector3[] _dots, System.Int32 _dotsCursor, System.Single _yaw, System.Int32 _segmentCnt) [0x00136] in <ee3f7728a4054543a91656140ca60f7c>:0 
  at BuildTool_BlueprintPaste.OperatingPrestage () [0x0016d] in <ee3f7728a4054543a91656140ca60f7c>:0 
  at BuildTool_BlueprintPaste._OnTick (System.Int64 time) [0x0004f] in <ee3f7728a4054543a91656140ca60f7c>:0 
  at BuildTool._GameTick (System.Int64 time) [0x000cf] in <ee3f7728a4054543a91656140ca60f7c>:0 
```

**Comment by innominata** [comment]
I've spent days on this bug and haven't come up with a solution yet. Will keep trying

**Comment by Wesmania** [comment]
If it helps you any, I was able to trigger the bug twice in the same way, so I might have a reliable way of triggering it.
* Start on a planet of size ~350-400.
* Place solar panels on the pole as following: one in the middle aligned with cardinal directions, then 4 adjacent in cardinal directions, then 4-5 full circles of solar panels, pressing '+' and '-' as appropriate for them to fit.
* Copy the solar panels then try to paste them on the opposite pole with an incorrect alignment (not aligned to cardinal directions).

**Comment by innominata** [comment]
Fixed in 2.3.14 I hope
https://dsp.thunderstore.io/package/download/Galactic_Scale/GalacticScale_Experimental/2.3.14/
